src/docs-builder/Cli/Move.cs

```diff
@@ -0,0 +1,228 @@
+		{
+			var originalPath = match.Value.Substring(match.Value.IndexOf('(') + 1, match.Value.LastIndexOf(')') - match.Value.IndexOf('(') - 1);
+
+			var newPath = originalPath;
+			var isAbsoluteStylePath = originalPath.StartsWith('/');
+			if (!isAbsoluteStylePath)
+			{
+				var targetDirectory = Path.GetDirectoryName(targetPath)!;
+				var sourceDirectory = Path.GetDirectoryName(sourceMarkdownFile.FilePath)!;
+				var fullPath = Path.GetFullPath(Path.Combine(sourceDirectory, originalPath));
+				var relativePath = Path.GetRelativePath(targetDirectory, fullPath);
+				newPath = relativePath;
+			}
+			var newLink = $"[{match.Groups[1].Value}]({newPath})";
+			var lineNumber = sourceContent.Substring(0, match.Index).Count(c => c == '\n') + 1;
+			var columnNumber = match.Index - sourceContent.LastIndexOf('\n', match.Index);
+			_logger.LogInformation(
+				string.Format(
+					ChangeFormatString,
+					match.Value,
+					newLink,
+					sourceMarkdownFile.SourceFile.FullName,
+					lineNumber,
+					columnNumber
+				)
+			);
+			return newLink;
+		});
+
+		_changes.Add((sourceMarkdownFile.FilePath, sourceContent, change));
+
+		foreach (var (_, markdownFile) in documentationSet.MarkdownFiles)
+		{
+			await ProcessMarkdownFile(
+				sourcePath,
+				targetPath,
+				markdownFile,
+				ctx
+			);
+		}
+
+		if (isDryRun)
+			return 0;
+
+		var targetDirectory = Path.GetDirectoryName(targetPath);
+		fileSystem.Directory.CreateDirectory(targetDirectory!);
+		fileSystem.File.Move(sourcePath, targetPath);
+		try
+		{
+			foreach (var (filePath, _, newContent) in _changes)
+				await fileSystem.File.WriteAllTextAsync(filePath, newContent, ctx);
+		}
+		catch (Exception)
+		{
+			foreach (var (filePath, originalContent, _) in _changes)
+				await fileSystem.File.WriteAllTextAsync(filePath, originalContent, ctx);
+			fileSystem.File.Move(targetPath, sourcePath);
+			throw;
+		}
+		return 0;
+	}
+
+	private bool ValidateInputs(string? source, string? target)
+	{
+
+		if (string.IsNullOrEmpty(source))
+		{
+			_logger.LogError("Source path is required");
+			return false;
+		}
+
+		if (string.IsNullOrEmpty(target))
+		{
+			_logger.LogError("Target path is required");
+			return false;
+		}
+
+		if (!Path.GetExtension(source).Equals(".md", StringComparison.OrdinalIgnoreCase))
+		{
+			_logger.LogError("Source path must be a markdown file. Directory paths are not supported yet");
+			return false;
+		}
+
+		if (!Path.GetExtension(target).Equals(".md", StringComparison.OrdinalIgnoreCase))
+		{
+			_logger.LogError("Target path must be a markdown file. Directory paths are not supported yet");
+			return false;
+		}
+
+		if (!fileSystem.File.Exists(source))
+		{
+			_logger.LogError($"Source file {source} does not exist");
+			return false;
+		}
+
+		if (fileSystem.File.Exists(target))
+		{
+			_logger.LogError($"Target file {target} already exists");
+			return false;
+		}
+
+		return true;
+	}
+
+	private async Task ProcessMarkdownFile(
```

Comment:
I was hoping that we could leverage markdig here. 

E.g we parse the document AST and include a linkrewriter parser (or make it part of our existing `DiagnosticLinksParser`. To mutate the links.

Then we can write the AST back out again as markdown. 

markdig supports roundtipping like this.

Reply:
Makes totally sense.

Given the time and the need for it already today/tomorrow, do you think we can leave it this way for now and refactor it in a follow-up using Markdig?

Reply:
100%! progress over perfection.